cot/src/config.rs

```diff
@@ -137,48 +211,164 @@ fn default_auth_backend() -> Arc<dyn AuthBackend> {
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use cot::config::{ProjectConfig, SecretKey};
+    ///
+    /// let config = ProjectConfig::builder()
+    ///     .secret_key(SecretKey::new(&[1, 2, 3]))
+    ///     .build();
+    /// assert_eq!(config.secret_key().as_bytes(), &[1, 2, 3]);
+    /// ```
     #[must_use]
     pub fn secret_key(&self) -> &SecretKey {
         &self.secret_key
     }
 
+    /// Get the fallback secret keys stored in the project configuration.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use cot::config::{ProjectConfig, SecretKey};
+    ///
+    /// let config = ProjectConfig::builder()
+    ///     .secret_key(SecretKey::new(&[1, 2, 3]))
+    ///     .fallback_secret_keys(vec![SecretKey::new(&[4, 5, 6])])
+    ///     .build();
+    /// assert_eq!(config.fallback_secret_keys(), &[SecretKey::new(&[4, 5, 6])]);
+    /// ```
     #[must_use]
     pub fn fallback_secret_keys(&self) -> &[SecretKey] {
         &self.fallback_secret_keys
     }
 
+    /// Get the authentication backend stored in the project configuration.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use cot::auth::UserId;
+    /// use cot::request::{Request, RequestExt};
+    /// use cot::response::Response;
+    ///
+    /// fn index(request: Request) -> cot::Result<Response> {
+    ///     let user = request
+    ///         .project_config()
+    ///         .auth_backend()
+    ///         .get_by_id(&request, UserId::Int(123));
+    ///
+    ///     // ... do something with the user
+    ///     # todo!()
+    /// }
+    /// ```
     #[must_use]
     pub fn auth_backend(&self) -> &dyn AuthBackend {
         &*self.auth_backend
     }
 
+    /// Get the database configuration stored in the project configuration.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use cot::config::ProjectConfig;
+    ///
+    /// let config = ProjectConfig::builder().build();
+    /// assert_eq!(config.database_config().url(), "sqlite::memory:");
+    /// ```
     #[must_use]
     #[cfg(feature = "db")]
     pub fn database_config(&self) -> &DatabaseConfig {
         &self.database_config
     }
 }
 
+/// A secret key.
+///
+/// This is a wrapper over a byte array, which is used to store a cryptographic
+/// key. This is useful for [`ProjectConfig::secret_key`] and
+/// [`ProjectConfig::fallback_secret_keys`], which are used to sign cookies and
+/// other sensitive data.
+///
+/// # Security
+///
+/// The implementation of the [`PartialEq`] trait for this type is constant-time
+/// to prevent timing attacks.
+///
+/// The implementation of the [`Debug`] trait for this type hides the secret key
+/// to prevent it from being leaked in logs or other debug output.
+///
+/// # Examples
+///
+/// ```
+/// use cot::config::SecretKey;
+///
+/// let key = SecretKey::new(&[1, 2, 3]);
+/// assert_eq!(key.as_bytes(), &[1, 2, 3]);
+/// ```
 #[repr(transparent)]
 #[derive(Clone)]
 pub struct SecretKey(Box<[u8]>);
```

Comment:
On a side note, why are we not using some safe storage for that and just keep that in plain~~text~~bytes?

Reply:
See #47. I don't think that makes much of a difference (as we need _some_ way to decode the value, so we can't make this more _secure_), so the best we can do is to make it slightly more annoying for attackers. Which is still valuable, but it's not a priority.